numpy/lib/_npyio_impl.py

```diff
@@ -1604,9 +1603,7 @@ def first_write(self, v):
                 else:
                     fmt = [fmt, ] * ncol
                 format = delimiter.join(fmt)
-            elif iscomplex_X and n_fmt_chars != (2 * ncol):
-                raise error
-            elif ((not iscomplex_X) and n_fmt_chars != ncol):
+            elif iscomplex_X and n_fmt_chars != (2 * ncol) or ((not iscomplex_X) and n_fmt_chars != ncol):
```

Comment:
The original is easier to read.

Reply:
Perhaps temporary variables with meaningful names would make it easier to read. I have to agree this looks less readable to me.